x-pack/plugin/core/src/internalClusterTest/java/org/elasticsearch/xpack/cluster/routing/allocation/DataTierAllocationDeciderIT.java

```diff
@@ -552,15 +552,21 @@ public String startColdOnlyNode(@Nullable String externalId) {
             nodeSettings.put(NODE_EXTERNAL_ID_SETTING.getKey(), externalId);
         }
 
-        return internalCluster().startNode(nodeSettings);
+        return startNodeAndWait(nodeSettings.build());
     }
 
     public void startFrozenOnlyNode() {
         Settings nodeSettings = Settings.builder()
             .putList("node.roles", Arrays.asList("master", "data_frozen", "ingest"))
             .put("node.attr.box", "frozen")
             .build();
-        internalCluster().startNode(nodeSettings);
+        startNodeAndWait(nodeSettings);
+    }
+
+    public String startNodeAndWait(Settings settings) {
+        var node = internalCluster().startNode(settings);
+        internalCluster().validateClusterFormed(); // autoManageMasterNodes is false so we must wait explicitly
```

Comment:
I am surprised that we need autoManageMasterNodes=false in this test

Reply:
It looks like we deliberately added it in #87735. I'm also not sure why, but didn't want to investigate right now.

Reply:
IIRC with autoManageMasterNodes=true the test might create master nodes that can contain data and that was problematic in the test scenarios that this test exercise.

Reply:
Hmm with `numDataNodes = 0` each test should start out with no nodes at all (it's a bad name) - this should give it control over the roles of each node it creates.